labels_command.php

```diff
@@ -0,0 +1,142 @@
+//
+// Do not prefix color with hash i.e. use 'cccccc' not '#cccccc'
+const LABELS_COLORS = [
+    'affects/v4' => '5319e7',
+    'affects/v5' => '0e8a16',
+    'complexity/low' => 'c2e0c6',
+    'complexity/medium' => 'fef2c0',
+    'complexity/high' => 'f9d0c4',
+    'Epic' => '3e4b9e',
+    'impact/low' => 'fef2c0',
+    'impact/medium' => 'f7c6c7',
+    'impact/high' => 'eb6420',
+    'impact/critical' => 'e11d21',
+    'rfc/accepted' => 'dddddd',
+    'rfc/draft' => 'dddddd',
+    'type/api-break' => '1d76db',
+    'type/bug' => 'd93f0b',
+    'type/docs' => '02d7e1',
+    'type/enhancement' => '0e8a16',
+    'type/userhelp' => 'c5def5',
+    'type/UX' => '006b75',
+    'type/other' => '975515',
+];
+
+// Rename existing labels 'from' => 'to'
+const LABELS_RENAME = [
+    'effort/easy' => 'complexity/low',
+    'effort/medium' => 'complexity/medium',
+    'effort/hard' => 'complexity/high',
+    'change/major' => 'type/api-break',
+    'type/api-change' => 'type/api-break',
+];
+
+$labelsCommand = function(InputInterface $input, OutputInterface $output): int {
+    // This is the code that is executed when running the 'labels' command
+
+    // variables
+    global $OUT, $REPOS_WITH_LABELS_UPDATED;
+    $OUT = $output;
+
+    // validate system is ready
+    validate_system();
+
+    // setup directories
+    setup_directories($input);
+
+    // CMS major version to use
+    $cmsMajor = $input->getOption('cms-major') ?: CURRENT_CMS_MAJOR;
+
+    // modules
+    $modules = filtered_modules($cmsMajor, $input);
+
+    // update labels
+    foreach ($modules as $module) {
+        $account = $module['account'];
+        $repo = $module['repo'];
+
+        // Fetch labels
+        $labels = github_api("https://api.github.com/repos/$account/$repo/labels");
+
+        foreach ($labels as $key => $label) {
+            // $label is an array, for example:
+            // 'id' => 427423377
+            // 'node_id' => MDU6TGFiZWw0Mjc0MjMzNzc="
+            // 'url' => "https://api.github.com/repos/silverstripe/silverstripe-config/labels/affects/v4"
+            // 'name' => "affects/v4"
+            // 'color' => "5319e7"
+            // 'default' => false
+            // 'description' => NULL
+            $url = $label['url'];
+            $name = $label['name']; // e.g. 'affects/v4'
+
+            // Rename label
+            // https://docs.github.com/en/rest/issues/labels#update-a-label
+            if (array_key_exists($name, LABELS_RENAME)) {
+                $newName = LABELS_RENAME[$name];
+                info("Updating label $name to $newName in $repo");
+                if ($input->getOption('dry-run')) {
+                    info('Not updating label on GitHub because --dry-run option is set');
+                } else {
+                    github_api($url, ['new_name' => $newName], 'PATCH');
+                }
+                $name = $newName;
+                $url = substr($url, 0, strlen($url) - strlen($name)) . $newName;
```

Comment:
```suggestion
                $url = preg_replace('#' . preg_quote($name, '#') . '$#', $newName', $url);
```
This is really hard to see what it's doing, but I think it was just replacing the old name at the end of the string with the new one? Please either make this change or add a comment that explains what the code does so we don't have to go through and parse it all each time we see it.

Reply:
Added comment